packages/devtools_app/lib/src/flutter/screen.dart

```diff
@@ -48,6 +57,17 @@ abstract class Screen {
   ///  * 'package:provider/'
   final String conditionalLibrary;
 
+  /// Screen controller for conditional screens.
+  ///
+  /// When non-null, this controller will be used for processing data from
+  /// offline imports. In order for conditional screens to have offline support,
+  /// the screen content returned by [build] needs to use this controller to
+  /// manage data.
+  ///
+  /// If null, we will use the corresponding controller from
+  /// [ProvidedControllers.offline] to process offline data for this screen.
+  final OfflineControllerMixin conditionalController;
```

Comment:
why do offline controllers for conditinoal imports need to be provided earlier than other controllers? It would be nice to work around that issue if practical instead of special casing how the offline controller needs to be provided for the conditional use case.